.travis.yml

```diff
@@ -24,15 +21,44 @@ cache:
 
 language: java
 
-env:
-  # TODO: JDK6 does not react well to concurrent usage of in-process javac as used by zinc, so
-  # we limit concurrency there.
-  - JDK=openjdk6   WORKERS=1
-  - JDK=openjdk7   WORKERS=6
-  - JDK=oraclejdk7 WORKERS=6
-  - JDK=oraclejdk8 WORKERS=6
-
-script: |
-  jdk_switcher use ${JDK}
-  ./pants --compile-zinc-worker-count=${WORKERS} doc test ::
+x-test: &x-test
+  script: |
+    java -version
+    ./pants --compile-zinc-worker-count=${WORKERS:-6} doc test ::
+
+x-switcher: &x-switcher
+  <<: *x-test
+  install:
+    - jdk_switcher use ${JDK}
+
+x-installer: &x-installer
+  <<: *x-test
+  before_install:
+    - wget https://github.com/sormuras/bach/raw/master/install-jdk.sh
```

Comment:
This is a bit scary. Would it be worth pointing at a particular SHA for now? Or can we include a comment referencing why we're not worried about this?

Reply:
Agreed - pinned.